doc/source/whatsnew/v1.5.1.rst

```diff
@@ -23,7 +23,8 @@ Fixed regressions
 
 Bug fixes
 ~~~~~~~~~
--
+- Bug in :class:`CSSToExcelConverter` leading to error when unrecognized ``border-style`` (including ``"hair"``) provided to Excel writers (:issue:`48649`)
```

Comment:
I would slightly learn toward moving this to `v1.6.0.rst` as it appears this was "unsupported" rather than "use to work"

Reply:
I could support either, but to play devil's advocate: pandas never specified that these particular border-styles were supported or unsupported. The closest thing to documentation was this comment block in `_border_style` which implies there is a path to getting the "hair" style when one didn't actually exist:
```
        # convert styles and widths to openxml, one of:
        #       'dashDot'
        #       'dashDotDot'
        #       'dashed'
        #       'dotted'
        #       'double'
        #       'hair'
        #       'medium'
        #       'mediumDashDot'
        #       'mediumDashDotDot'
        #       'mediumDashed'
        #       'slantDashDot'
        #       'thick'
        #       'thin'
```

I think the error when using `xlsxwriter` with an unsupported border-style should be fixed in v1.5.1 (i.e. `return "none"` at end of function), as the stack location and error message make it very unclear where the user has introduced the error. And if we're fixing that, then it's pretty small step from there to add the support for the other styles.

I definitely think I'm onboard with removing the color defaulting portion from a v1.5.1 and moving it to a separate PR for v1.6.0

Reply:
> I think the error when using `xlsxwriter` with an unsupported border-style should be fixed in v1.5.1 (i.e. `return "none"` at end of function), as the stack location and error message make it very unclear where the user has introduced the error.

The criteria for whether something should go in a patch or minor version is "did it work before". That is, patch releases should only fix regressions. As this isn't a regression, I'm also +1 on putting this in 1.6.0.

Reply:
I suggest that if `border-style` attribute was supported for some styles, in v1.5.0, one should, by extension, assume that all the different `border-style` values were supported **without** their explicit documentation. And a failing to not support certain values is either a bug in the code or a bug in the documentation. I support 1.5.1.

Reply:
Okay given the reduced scope (removing the color defaulting). I would be okay with 1.5.1